src/libsyntax/parse/parser.rs

```diff
@@ -4719,6 +4719,10 @@ impl<'a> Parser<'a> {
         if !self.eat(&token::OpenDelim(token::Brace)) {
             let sp = self.span;
             let tok = self.this_token_to_string();
+            let mut do_not_suggest_help = false;
+            if self.token.is_keyword(keywords::In) || self.token == token::Colon {
+                do_not_suggest_help = true;
+            }
             let mut e = self.span_fatal(sp, &format!("expected `{{`, found `{}`", tok));
```

Comment:
While you're doing this, could you add this here?

```
e.span_label(sp, "expected `{`");
```

Reply:
@estebank sure!